app/controllers/users_controller.rb

```diff
@@ -2,8 +2,36 @@ class UsersController < ApplicationController
   after_action :verify_authorized
   after_action :verify_policy_scoped, only: :index
 
+  rescue_from ActiveRecord::RecordNotFound do
+    render template: "errors/not_found", status: :not_found
+  end
+
   def index
     authorize current_user, :can_manage_user?
     render template: "users/index", locals: { users: policy_scope(User) }
   end
+
+  def edit
+    authorize current_user, :can_manage_user?
+    @user = User.find(params[:id])
+  end
+
+  def update
+    authorize current_user, :can_manage_user?
+
+    @user = User.find(params[:id])
+    @user.role = user_params[:role]
+
+    if @user.save
+      redirect_to users_path
+    else
+      render :edit
+    end
```

Comment:
Can we memorise `@user` something like https://github.com/alphagov/forms-admin/pull/372/files#diff-cfdccd0a9d5df5a43aaad2a35d36ebbe187c52ad5fdc9846fa189d04537adb6eR8-R15
?